### DISCUSSION TITLE
Can cloud Hypervisor be built without KVM

### DISCUSSION THREAD

**Discussion by lonnietc** [cloud-hypervisor/cloud-hypervisor#3699] [Q&A]
Hello,

I am researching some work on a new type of kernel and OS that does not have any type of KVM and need a small-footprint and fast VMM to run on it which made me wonder if Cloud-Hypervisor could be compiled and ported to the new system in some way?

Any advice or suggestions would be greatly appreciated.
Thanks in advance.

**Comment by sboeuf** [comment]
No you can't run Cloud Hypervisor without a KVM or MSHV support on your host OS.

**Comment by liuw** [comment]
Speaking of porting, you can have a look at the hypervisor crate in the tree.

**Comment by lonnietc** [comment]
@liuw Thanks as that might be a good step forward.  I am thinking that I might need to try and write up some type of kvm "shim" layer for the new kernel so that from that point on, Cloud Hypervisor, Qemu-Kvm, etc. would not know the difference.

I am sure that would be a LOT of work, but may be required if I want to use one of these VMM's on the new kernel.

Have a great day.
